classes/class3/homework3/routes/cats.js

```diff
@@ -0,0 +1,97 @@
+var mongoose = require('mongoose');
+
+//Define the properties of our cats
+var catSchema = mongoose.Schema({
+	name: String, age: Number, colors: [String]
+});
+var Cat = mongoose.model('Cat', catSchema);
+
+//These are the possible names
+var names = ["Whiskers", "Skittles", "Spots", "Tinky", "Sophie", "Agatha", "Dorothy", "Taz", "Oreo", "Muffin"];
+
+//These are the possible colors
+var colors = ["black", "white", "tabby", "orange", "golden"];
+
+var catList = function(req, res) {
+	//List all of the cats currently stored in the database
+	Cat.find({}, function(err, catList) {
+		if (err) {
+			console.error("Couldn't find cat list: ", err);
+		}
+		res.render('cats', {'cats': catList});
+	});
+};
+
+var addCat = function(req, res) {
+	//Choose a random name from the possible names
+	var catName = names[Math.floor(Math.random()*names.length)];
+
+	//Shuffle the cat array
+	colors.sort(function() {
```

Comment:
Pretty interesting way to get a random color, totally works :+1: 
Keep in mind that if colors was long, it can get a little expensive to shuffle the whole thing, easy just to grab a random number.